assets/js/components/buy-checkout.component.js

```diff
@@ -101,6 +97,22 @@ function BuyCheckoutController ($rootScope, $scope, $timeout, $q, currency, Wall
     }
   };
 
+  $scope.enableBuy = () => {
+    let obj = {
+      'BTC Order': $scope.format($scope.fromSatoshi(state.btc || 0, $scope.bitcoin), $scope.bitcoin, true),
+      'Payment Method': this.buyAccount.accountType + ' (' + this.buyAccount.accountNumber + ')',
+      'TOTAL_COST': $scope.format($scope.fromSatoshi(state.total || 0, $scope.dollars), $scope.dollars, true)
+    };
+
+    $scope.formattedTrade = formatTrade.confirm(obj);
+
+    $uibModal.open({
+      templateUrl: 'partials/confirm-trade-modal.pug',
+      windowClass: 'bc-modal trade-summary',
+      scope: $scope
```

Comment:
Is there anyway to avoid passing $scope to the modal? Sharing scopes tends to blur interfaces and doesn't lend well to reusability.